src/nvim/screen.c

```diff
@@ -4548,7 +4548,7 @@ static void screen_line(int row, int coloff, int endcol, int clear_width, int rl
       int c;
 
       c = fillchar_vsep(&hl);
-      if (ScreenLines[off_to] != c
+      if (ScreenLines[off_to] != (schar_T)c
```

Comment:
For consistency with how upstream patches are merged, lets leave 567 in this PR and I will remove the commit from my branch.